File: AOSharp.UI/src/pluginSections.ts
import type { Plugin } from './types';

export const PLUGIN_SECTIONS = ['Library', 'Utility', 'Combat', 'Bots', 'Other'] as const;
export type PluginSection = (typeof PLUGIN_SECTIONS)[number];

export function getPluginSection(plugin: Plugin): PluginSection {
  if (plugin.isLibrary) return 'Library';
  const s = plugin.section?.trim();
  if (s && (PLUGIN_SECTIONS as readonly string[]).includes(s)) return s as PluginSection;
  return 'Other';
}

File: AOSharp/Models/PluginSections.cs
using System;

namespace AOSharp.Models
{
    public static class PluginSections
    {
        public const string Library = "Library";
        public const string Utility = "Utility";
        public const string Combat = "Combat";
        public const string Bots = "Bots";
        public const string Other = "Other";

        public static readonly string[] DisplayOrder =
        {
            Library,
            Utility,
            Combat,
            Bots,
            Other
        };

        public static string Normalize(string section)
        {
            if (string.IsNullOrWhiteSpace(section))
                return Other;

            foreach (var valid in DisplayOrder)
            {
                if (string.Equals(valid, section.Trim(), StringComparison.OrdinalIgnoreCase))
                    return valid;
            }

            return Other;
        }
    }
}
